### DISCUSSION TITLE
I'm using .Skip(1) a lot to avoid unwanted values during initialization, am I doing it wrong?

### DISCUSSION THREAD

**Discussion by bonzaiferroni** [reactiveui/ReactiveUI#3115] [Q&A]
I've had several bugs that came down to values emitted during initialization that shouldn't be emitted. For example, a variable with the `[Reactive]` attribute that is subscribed to using `this.WhenAnyValue(x => x.MyProperty)` (I'm using `ReactiveUI.Fody`). In many cases the first value is actually not valid so each subscriber must use `.Skip(1)`. Is that the correct way to handle it or is there a better way?

**Comment by geometrikal** [comment]
Yes I also have the first value either being the default value or null during initialization. I either filter / check for null or use `.Skip(1)`.

**Comment by glennawatson** [comment]
WhenAnyValue will emit the current value always unless you have A skip(1). So if it is null you'll get a null emission straight away.

**Comment by anaisbetts** [comment]
This is one disadvantage of Fody, if you did it the non-Fody way, you could just initialize the backing field

```cs
string _SomeBackingField = "My First Value";
public SomeBackingField {
  get => _SomeBackingField;
  set => this.RaiseAndSetIfChanged(...);
}
```
